compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert_slice.mlir

```diff
@@ -68,3 +69,42 @@ func.func @insert_slice_within_dispatch_workgroups_not_converted() -> tensor<f32
   }
   return %0 : tensor<f32>
 }
+
+// -----
+
+func.func @insert_slice_convert_dynamic_offset_and_size
+    (%arg0 : tensor<?x24x48xf32>, %arg1 : tensor<1x?x48xf32>, %offset : index, %size : index) ->
+    tensor<?x24x48xf32> {
+  %c0 = arith.constant 0 : index
+  %0 = tensor.insert_slice %arg1 into %arg0[%offset, 2, 0] [1, %size, 48] [1, 1, 1] :
+      tensor<1x?x48xf32> into tensor<?x24x48xf32>
+  return %0 : tensor<?x24x48xf32>
+}
+// CHECK-LABEL: func.func @insert_slice_convert
+//  CHECK-SAME:   %[[ARG0:[a-zA-Z0-9_]+]]
+//  CHECK-SAME:   %[[ARG1:[a-zA-Z0-9_]+]]
+//  CHECK-SAME:   %[[ARG2:[a-zA-Z0-9_]+]]
+//  CHECK-SAME:   %[[ARG3:[a-zA-Z0-9_]+]]
+//   CHECK-DAG:   %[[C0:.+]] = arith.constant 0
+//   CHECK-DAG:   %[[C2:.+]] = arith.constant 2
+//   CHECK-DAG:   %[[DIM0:.+]] = tensor.dim %[[ARG0]], %[[C0]]
+//       CHECK:   %[[UPDATE:.+]] = flow.tensor.update %[[ARG1]], %[[ARG0]][%[[ARG2]], %[[C2]], %[[C0]]]
+//  CHECK-SAME:     : tensor<1x?x48xf32>{%[[ARG3]]} -> %[[ARG0]] as tensor<?x24x48xf32>{%[[DIM0]]}
+
+// -----
+
+// CHECK-LABEL: func.func @insert_slice_dynamic_tensor_result_not_converted
+func.func @insert_slice_dynamic_tensor_result_not_converted
+    (%arg0 : tensor<?x24x48xf32>, %arg1 : tensor<1x4x48xf32>, %offset : index) ->
+    tensor<?x24x48xf32> {
+  %x = arith.constant 100 : index
+  %0 = flow.dispatch.workgroups[%x]() : () -> (tensor<i64>) = () {
+    flow.return
+  }
+  %idx_i64 = tensor.extract %0[] : tensor<i64>
+  %idx = arith.index_cast %idx_i64 : i64 to index
+  // CHECK-NOT: flow.tensor.update
```

Comment:
what is it supposed to be? it's a good idea to CHECK-NOT what you don't want and CHECK what you do in cases where something else is produced instead. is it assumed to be tensor.insert_slice still?